PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/identity/ad/ui/RegisterApplicationInAzureAdDialog.kt

```diff
@@ -0,0 +1,176 @@
+            val originalDomain : String,
+            val originalCallbackUrl : String,
+            val originalIsMultiTenant : Boolean,
+            var allowOverwrite : Boolean
+    ) {
+        var updatedDisplayName: String = originalDisplayName
+        var updatedDomain : String = originalDomain
+        var updatedClientId : String = originalClientId
+        var updatedCallbackUrl : String = originalCallbackUrl
+        var updatedIsMultiTenant : Boolean = originalIsMultiTenant
+        var hasChanges : Boolean = false
+    }
+
+    private val panel = panel {
+
+        noteRow(RiderAzureBundle.message("dialog.identity.ad.register_app.description"))
+
+        row {
+            val label = JLabel(RiderAzureBundle.message("dialog.identity.ad.register_app.display_name"))
+            label()
+
+            cell {
+                textField({ model.originalDisplayName },
+                        {
+                            model.updatedDisplayName = it
+                            model.hasChanges = true
+                        })
+                        .withValidationOnInput { validateNotEmpty(it) }
+                        .component.apply { label.labelFor = this }
+            }
+        }
+
+        row {
+            val label = JLabel(RiderAzureBundle.message("dialog.identity.ad.register_app.callback_url"))
+            label()
+
+            cell {
+                textField({ model.originalCallbackUrl },
+                        {
+                            model.updatedCallbackUrl = it
+                            model.hasChanges = true
+                        })
+                        .withValidationOnInput { validateNotEmpty(it) ?: validateUrl(it) }
+                        .component.apply { label.labelFor = this }
+            }
+        }
+
+        row {
+            val label = JLabel(RiderAzureBundle.message("dialog.identity.ad.register_app.domain"))
+            label()
+
+            cell {
+                textField({ model.originalDomain },
+                        {
+                            model.updatedDomain = it
+                            model.hasChanges = true
+                        })
+                        .withValidationOnInput { validateNotEmpty(it) }
+                        .component.apply { label.labelFor = this }
+            }
+        }
+
+        row {
+            cell {
+                checkBox(RiderAzureBundle.message("dialog.identity.ad.register_app.is_multi_tenant"),
+                        { model.originalIsMultiTenant },
+                        {
+                            model.updatedIsMultiTenant = it
+                            model.hasChanges = true
+                        })
+            }
+        }
+
+        hideableRow(RiderAzureBundle.message("dialog.identity.ad.register_app.more_options")) {
+            row {
+                val label = JLabel(RiderAzureBundle.message("dialog.identity.ad.register_app.client_id"))
```

Comment:
Do we need to add validation for Client ID. I'm getting `Status code 400, "Unsupported or invalid query filter clause specified for property 'appId' of resource 'Application'."` error when Client ID is not a valid Guid.

Reply:
Fixed.